data-service-client/build.gradle

```diff
@@ -1,6 +1,8 @@
-description 'Blox DataService Client'
+plugins {
+    id "java"
```

Comment:
Answering my own question after [researching this for the spring dependency plugin](https://plugins.gradle.org/plugin/io.spring.dependency-management). The biggest problem for our case is:

> **Cannot be used in conjunction with `subprojects {}`, `allprojects {}` etc.**
> It is not possible to use the familiar pattern of applying a plugin to multiple projects at once using `subprojects {}` etc. at the moment. There is currently no mechanism for applying a plugin to multiple projects at once. At the moment, each project that requires a plugin must declare so in the `plugins {}` block in its buildscript.
>
> Future versions of Gradle will remove this restriction.
>
> If the current limitations of the new plugin mechanism are problematic for your build, the recommended approach is to use the “traditional” mechanism for the time being.

Since we're doing this for the spotless plugin, among others, it's probably not going to work to go this way fully right now, unless we want to duplicate the config for that across all projects.